server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java

```diff
@@ -139,25 +151,52 @@ public static void download(Repository repository, ShardId shardId, ThreadPool t
             shardId,
             fileTransferTracker
         );
+
+        // Dummy stats tracker to ensure the flow doesn't break on first download of tlog files.
+        // This first download is to ensure the segment file corruption checks pass.
+        // These tlog files will get downloaded again for the actual tlog replay.
+        translogTransferManager.setRemoteTranslogTracker(new RemoteTranslogTracker(shardId, 1000, 1000, 1000, 1000, 1000, 1000));
         RemoteFsTranslog.download(translogTransferManager, location, logger);
     }
 
     public static void download(TranslogTransferManager translogTransferManager, Path location, Logger logger) throws IOException {
         logger.trace("Downloading translog files from remote");
+        RemoteTranslogTracker statsTracker = translogTransferManager.getRemoteTranslogTracker();
+        long bytesBefore = statsTracker.getDownloadBytesSucceeded();
+        long downloadStartTime = RemoteStoreUtils.getCurrentSystemNanoTime();
         TranslogTransferMetadata translogMetadata = translogTransferManager.readMetadata();
         if (translogMetadata != null) {
             if (Files.notExists(location)) {
                 Files.createDirectories(location);
             }
+
             // Delete translog files on local before downloading from remote
             for (Path file : FileSystemUtils.files(location)) {
                 Files.delete(file);
             }
+
             Map<String, String> generationToPrimaryTermMapper = translogMetadata.getGenerationToPrimaryTermMapper();
             for (long i = translogMetadata.getGeneration(); i >= translogMetadata.getMinTranslogGeneration(); i--) {
                 String generation = Long.toString(i);
                 translogTransferManager.downloadTranslog(generationToPrimaryTermMapper.get(generation), generation, location);
             }
+
+            long downloadEndTime = RemoteStoreUtils.getCurrentSystemNanoTime();
+            long downloadEndTimeMs = System.currentTimeMillis();
+            long durationInMillis = (downloadEndTime - downloadStartTime) / 1_000_000L;
+            long bytesDownloaded = statsTracker.getDownloadBytesSucceeded() - bytesBefore;
+
+            statsTracker.setLastSuccessfulDownloadTimestamp(downloadEndTimeMs);
+            // We update the duration at the end of successfully downloading all of metadata, .tlog, .ckp
+            // files because this is not a file-level metric but a sync-level metric.
+            // This also ensures the bytes per sec moving average can be correlated.
+            statsTracker.addDownloadTimeInMillis(durationInMillis);
+            statsTracker.updateDownloadBytesMovingAverage(bytesDownloaded);
+            statsTracker.updateDownloadTimeMovingAverage(durationInMillis);
+            if (durationInMillis > 0) {
+                statsTracker.updateDownloadBytesPerSecMovingAverage(bytesDownloaded * 1_000L / durationInMillis);
+            }
```

Comment:
This entire block can be moved to tracker by providing required params like `downloadStartTime`, `bytesBefore` etc.

Reply:
Addressed. Resolving this thread. Please reopen if any changes are needed.